### ISSUE TITLE
Add DatetimeOrdinal

### ISSUE THREAD

**Issue by michaelrussell4** [feature-engine/feature_engine#818]
It'd be nice to have the ability to convert datetime columns to ordinal values. The DatetimeFeatures is nice for when one wants to extract the year, month, day, etc., but sometimes it's desirable to have the date simply as an ordinal value.

Here's an example I've used in code before:

```py
class DatetimeOrdinal(BaseEstimator, TransformerMixin):
    def fit(self, X, y=None):
        return self

    def transform(self, X):
        return X.apply(lambda x: x.map(pd.Timestamp.toordinal, na_action='ignore'))

    def inverse_transform(self, X):
        return X.apply(lambda x: x.map(pd.Timestamp.fromordinal, na_action='ignore'))
```

**Comment by solegalli** [comment]
Thank you @michaelrussell4 

I wasn't aware of this functionality.

When would it be useful to map dates to ordinal numbers? Do I understand correctly that the cardinality of the variable will still be high after this representation?

**Comment by michaelrussell4** [comment]
Here are some reasons why this method can be useful:

- **Preserves Temporal Distance**: Retains the actual time intervals between dates, useful for models that benefit from continuity.
- **Simplified Feature Space**: Reduces complexity by converting datetime into a single numerical value.
- **Effective for Linear Trends**: Ideal for models that assume linear relationships over time, such as regression.
- **Captures Long-Term Trends**: Better suited for datasets where long-term changes are more relevant than short-term cycles.

The high cardinality of this method does have to be accounted for, probably best with a max-min normalisation (British spelling just for you 😉).

This technique isn't as popular as the DatetimeFeature extraction you currently have but I think it'd be worth considering adding.

**Comment by solegalli** [comment]
Thank you @michaelrussell4 

We'll keep it on the radar :)

If you want to give it a go, you are welcome!

**Comment by michaelrussell4** [comment]
I surely will if I get a chance. Thanks for your work!

**Comment by VascoSch92** [comment]
Is that still a thing? Can I try to give a shot?

**Comment by michaelrussell4** [comment]
> Is that still a thing? Can I try to give a shot?

I think it would be a useful contribution 🙂

**Comment by VascoSch92** [comment]
Ok. If I understand, you want something that given the data frame

```text
       date1      date2
0 2023-01-01 2024-01-01
1 2023-02-01 2025-03-01
2        NaT 2026-05-01
```

transform it into

```text
 date1   date2
0  738521  738886
1  738552  739311
2     NaT  739737
```

A question: To avoid the high cardinality problem, would you like to provide a `start_date` parameter for the encoder to start counting from 1?

For example, take the above df and use as `start_date = "2023-01-01"`. Then you have

```text
Original DataFrame:
       date1      date2
0 2023-01-01 2024-01-01
1 2023-02-01 2025-03-01
2        NaT 2026-05-01

Transformed DataFrame (Ordinal):
  date1  date2
0     1    366
1    32    791
2   NaT   1217
```

**Comment by michaelrussell4** [comment]
Yeah I think that's a great optional parameter to add. Then people can have the option of doing that or just handling it later with a scalar transformer.

And everything else looks just how it ought to.

**Comment by VascoSch92** [comment]
I have already a first draft ready.

@solegalli what do you think?
Should I open a PR (draft) to discuss it?

**Comment by VascoSch92** [comment]
@solegalli :-) ?
